packages/web-components/fast-foundation/src/data-grid/data-grid.ts

```diff
@@ -520,17 +507,48 @@ export class FASTDataGrid extends FASTElement {
                     this.focusOnCell(
                         this.rowElements.length - 1,
                         this.columnDefinitions.length - 1,
-                        true
+                        "nearest"
                     );
                 }
                 break;
         }
     }
 
+    private getPageSize(): number {
+        if (this.pageSize) {
+            return this.pageSize;
+        }
+
+        let rowHeight = 50;
+        this.rowElements.forEach(element => {
+            if (
+                !element.hasAttribute("rowType") ||
+                !element.getAttribute("rowType")?.includes("header")
+            ) {
+                rowHeight = element.clientHeight;
+            }
+        });
+
+        let pageSize: number = 1;
+
+        if (rowHeight === 0) {
+            return pageSize;
+        }
+
+        if (this.clientHeight < this.scrollHeight) {
+            pageSize = this.clientHeight / rowHeight;
+        } else {
+            pageSize = document.body.clientHeight / rowHeight;
+        }
+
+        pageSize = Math.max(Math.round(pageSize), 1);
+        return pageSize;
+    }
+
     private focusOnCell = (
         rowIndex: number,
         columnIndex: number,
-        scrollIntoView: boolean
+        alignment: "start" | "center" | "end" | "nearest"
```

Comment:
What does "nearest" mean in this context? I'm not following that

Reply:
It's the parameters for the "block" parameter of a "scrollIntoView" call.  I changed the param in the `focusOnCell` function to be the actual system type of `ScrollLogicalPosition' to make it clearer. 
https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView
`